features/steps/steps.go

```diff
@@ -1,13 +1,263 @@
+	url := fmt.Sprintf(
+		"/datasets/%s/editions/%s/versions/%s",
+		datasetID,
+		edition,
+		version,
+	)
+
+	c.DatasetAPI.NewHandler().
+		Get(url).
+		Reply(http.StatusOK).
+		BodyString(v.Content)
+
+	return nil
 }
 
-func (c *Component) iShouldReceiveAHelloworldResponse() error {
+// theFollowingVersionIsUpdated generates a mocked response for dataset API
+// PUT /instances/{id} with the provided instance response
+func (c *Component) theFollowingVersionWillBeUpdated(datasetID, edition, version string, v *godog.DocString) error {
+	url := fmt.Sprintf(
+		"/datasets/%s/editions/%s/versions/%s",
+		datasetID,
+		edition,
+		version,
+	)
+
+	c.DatasetAPI.NewHandler().
+		Put(url).
+		AssertCustom(newPutVersionAssertor([]byte(v.Content))).
+		Reply(http.StatusOK)
+
+	return nil
+}
+
+func (c *Component) thisCantabularMetadataExportEventIsConsumed(input *godog.DocString) error {
+	ctx := context.Background()
+
+	// testing kafka message that will be produced
+	var testEvent event.CSVCreated
+	if err := json.Unmarshal([]byte(input.Content), &testEvent); err != nil {
+		return fmt.Errorf("error unmarshaling input to event: %w body: %s", err, input.Content)
+	}
+
+	log.Info(ctx, "event to marshal: ", log.Data{
+		"event": testEvent,
+	})
+
+	// marshal and send message
+	b, err := schema.CSVCreated.Marshal(testEvent)
+	if err != nil {
+		return fmt.Errorf("failed to marshal event from schema: %w", err)
+	}
+
+	log.Info(ctx, "marshalled event: ", log.Data{
+		"event": b,
+	})
+
+	c.producer.Channels().Output <- b
+	return nil
+}
+
+func (c *Component) theFollowingFileCanBeSeenInMinio(fileName string, bucket string) error {
+	ctx := context.Background()
+
+	var b []byte
+	f := aws.NewWriteAtBuffer(b)
+
+	// probe bucket with backoff to give time for event to be processed
+	retries := 5
+	timeout := 1
+	var numBytes int64
+	var err error
+
+	for {
+		if numBytes, err = c.S3Downloader.Download(f, &s3.GetObjectInput{
+			Bucket: aws.String(bucket),
+			Key:    aws.String(fileName),
+		}); err == nil || retries <= 0 {
+			break
+		}
+
+		retries--
+
+		log.Info(ctx, "error obtaining file from minio. Retrying.", log.Data{
+			"error":        err,
+			"retries_left": retries,
+		})
+
+		time.Sleep(time.Second * time.Duration(timeout))
+		timeout *= 2
+	}
+	if err != nil {
+		return fmt.Errorf(
+			"error obtaining file from minio. Last error: %w",
+			err,
+		)
+	}
+
+	if numBytes < 1 {
+		return errors.New("file length zero")
+	}
+
+	log.Info(ctx, "got file contents", log.Data{
+		"contents": string(f.Bytes()),
+	})
+
+	return nil
+}
+
+func (c *Component) theseCSVWCreatedEventsShouldBeProduced(events *godog.Table) error {
+	expected, err := assistdog.NewDefault().CreateSlice(new(event.CSVWCreated), events)
+	if err != nil {
+		return fmt.Errorf("failed to create slice from godog table: %w", err)
+	}
+
+	var got []*event.CSVWCreated
+	listen := true
+
+	for listen {
+		select {
+		case <-time.After(time.Second * 1):
```

Comment:
I'd suggest using a constant for this timeout:
https://github.com/ONSdigital/dp-cantabular-csv-exporter/blob/be9bb7dee2bbe104536bac61befe7d15c24cd297/features/steps/steps.go#L154